### PR TITLE
[FLINK-4546] [table] Remove STREAM keyword in Stream SQL

### PR DESCRIPTION
**Pull request by wuchong**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

This PR remove STREAM keyword in Stream SQL and update the documentation.

### CODE REVIEW

**Comment by twalthr** [comment]
Thank @wuchong. Looks good to merge. @fhueske do you also want to have a look at it?

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/rules/FlinkRuleSets.scala [hunk 1]
```diff
@@ -112,7 +112,7 @@ object FlinkRuleSets {
   */
   val DATASTREAM_OPT_RULES: RuleSet = RuleSets.ofList(
 
-      RemoveDeltaRule.INSTANCE,
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/rules/FlinkRuleSets.scala]: The `RemoveDeltaRule` class can be removed as well.

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/schema/DataStreamTable.scala [hunk 2]
```diff
@@ -18,22 +18,11 @@
...
-import org.apache.flink.api.table.FlinkTypeFactory
 import org.apache.flink.streaming.api.datastream.DataStream
 
 class DataStreamTable[T](
     val dataStream: DataStream[T],
     override val fieldIndexes: Array[Int],
     override val fieldNames: Array[String])
   extends FlinkTable[T](dataStream.getType, fieldIndexes, fieldNames) {
-
-  override def getRowType(typeFactory: RelDataTypeFactory): RelDataType = {
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/schema/DataStreamTable.scala]: Why did you remove this method?

**Comment by fhueske** [comment]
Hi @wuchong, thanks for the PR. Just had two minor comments. Thanks, Fabian

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/schema/DataStreamTable.scala (hunk 2 above)
Comment (wuchong) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/plan/schema/DataStreamTable.scala]: Because this override method is almost the same with super method.  I remove it to keep consistent with `DataSetTable` which do not override the `getRowType` method.

**Comment by wuchong** [comment]
Hi @fhueske , thanks for reviewing,  I updated this PR.

**Comment by wuchong** [comment]
Hi @fhueske @twalthr , do you have time to have a look again? Any advices are welcome.

**Comment by twalthr** [comment]
It seems that Calcite is planning to add additional clauses to the SQL statement. Like `SELECT STREAM x FROM y EMIT xyz`. If we really remove the keyword we have to make sure that Calcite's parser will also parse `EMIT` without the `STREAM` keyword.

@fhueske what do you think?

**Comment by twalthr** [comment]
Thanks @wuchong. Will merge this.
